flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -85,65 +86,71 @@ public static TypeInfo toHiveTypeInfo(DataType dataType) {
 		LogicalTypeRoot type = dataType.getLogicalType().getTypeRoot();
 
 		if (dataType instanceof AtomicDataType) {
```

Comment:
How about using a `LogicalTypeVisitor` or `LogicalTypeDefaultVisitor` here instead? It would simplify the logic here because `if/else/instanceof` checks would not be required anymore. Furthermore, we would force implementers to make sure all cases are covered.

Reply:
Yep, that would be more elegant.